checkov/openapi/checks/resource/v2/BaseOpenapiCheckV2.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from typing import Iterable, Any, Dict, Tuple
+from abc import abstractmethod
+from checkov.openapi.checks.base_openapi_check import BaseOpenapiCheck
+from checkov.common.models.enums import CheckResult, CheckCategories
+
+
+class BaseOpenapiCheckV2(BaseOpenapiCheck):
+    def __init__(self, name: str, id: str, categories: Iterable[CheckCategories], supported_entities: Iterable[str],
+                 block_type: str) -> None:
+        super().__init__(
+            name=name,
+            id=id,
+            categories=categories,
+            supported_entities=supported_entities,
+            block_type=block_type,
+        )
+
+    @abstractmethod
+    def scan_openapi_conf(self, conf: Dict[str, Any], entity_type: str) -> Tuple[CheckResult, Dict[str, Any]]:
```

Comment:
```suggestion
    def scan_openapi_conf(self, conf: dict[str, Any], entity_type: str) -> tuple[CheckResult, dict[str, Any]]:
```
if add `from __future__ import annotations`, like you did, then you can use the built-in types for typing `Dict`, `List`, `Set` and `Tuple` and then of course you can remove the import of those types, otherwise `flake8` will blame you 😄